docs/notebooks/Calculating ground states on large scale systems with the quantum krylov subspace method.ipynb

```diff
@@ -0,0 +1,1037 @@
+    }
+   },
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Hadamard test\n",
```

Comment:
I would give a reference to the Hadamard test

I would suggest more description of what is going on. You also introduce $U_k$ and $P$ without stating what they are and how they related to what you are doing.

Reply:
Looking for a reference for the Hadamard test. Haven't found a neat one yet but I'll keep looking. 

$U_k$ is now defined in the introduction at the top while $P$ is introduced together with the equations that use it